src/tests/utests/iotc_utest_memory_calloc.c

```diff
@@ -0,0 +1,195 @@
+ *
+ * This is part of the Google Cloud IoT Device SDK for Embedded C.
+ * It is licensed under the BSD 3-Clause license; you may not use this file
+ * except in compliance with the License.
+ *
+ * You may obtain a copy of the License at:
+ *  https://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "iotc_macros.h"
+#include "iotc_tt_testcase_management.h"
+#include "tinytest.h"
+#include "tinytest_macros.h"
+
+#include "iotc.h"
+#include "iotc_allocator.h"
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+
+#ifndef IOTC_TT_TESTCASE_ENUMERATION__SECONDPREPROCESSORRUN
+
+#endif  // IOTC_TT_TESTCASE_ENUMERATION__SECONDPREPROCESSORRUN
+
+IOTC_TT_TESTGROUP_BEGIN(utest_memory_calloc)
+
+/* Happy cases. */
+IOTC_TT_TESTCASE(
+    utest__iotc_memory_calloc__num_1__size_8,
+    {
+      void* ptr = __iotc_calloc( /*num=*/1, /*size=*/8);
+      tt_want_ptr_op(NULL, !=, ptr);
+      __iotc_free(ptr);
+    })
+
+IOTC_TT_TESTCASE(
+    utest__iotc_memory_calloc___num_255__size_55,
+    {
+      void* ptr = __iotc_calloc( /*num=*/255, /*size=*/55);
+      tt_want_ptr_op(NULL, !=, ptr);
+      __iotc_free(ptr);
+    })
+
+IOTC_TT_TESTCASE(
+    utest__iotc_memory_calloc___num_1000000__size_88,
+    {
+      void* ptr = __iotc_calloc( /*num=*/1000000, /*size=*/88);
+      tt_want_ptr_op(NULL, !=, ptr);
+      __iotc_free(ptr);
+    })
+
+/* Edge cases. */
+IOTC_TT_TESTCASE(
+    utest__iotc_memory_calloc__num_0__size_8,
+    {
+      void* ptr = __iotc_calloc( /*num=*/0, /*size=*/8);
+      tt_want_ptr_op(NULL, ==, ptr);
+      __iotc_free(ptr);
+    })
+
+IOTC_TT_TESTCASE(
+    utest__iotc_memory_calloc__num_1__size_0,
+    {
+      void* ptr = __iotc_calloc( /*num=*/1, /*size=*/0);
+      tt_want_ptr_op(NULL, ==, ptr);
+      __iotc_free(ptr);
+    })
+
+IOTC_TT_TESTCASE(
+    utest__iotc_memory_calloc__num_0__size_0,
+    {
+      void* ptr = __iotc_calloc( /*num=*/0, /*size=*/0);
+      tt_want_ptr_op(NULL, ==, ptr);
+      __iotc_free(ptr);
+    })
+
+IOTC_TT_TESTCASE(
+    utest__iotc_memory_calloc__num_max__size_0,
+    {
+      void* ptr = __iotc_calloc( /*num=*/SIZE_MAX, /*size=*/0);
+      tt_want_ptr_op(NULL, ==, ptr);
+      __iotc_free(ptr);
+    })
+
+IOTC_TT_TESTCASE(
+    utest__iotc_memory_calloc__num_0__size_max,
+    {
+      void* ptr = __iotc_calloc( /*num=*/0, /*size=*/SIZE_MAX);
+      tt_want_ptr_op(NULL, ==, ptr);
+      __iotc_free(ptr);
+    })
+
+/* Edge cases - SIZE_MAX
+   Note: FreeRTOS heap size isn't large enough for SIZE_MAX allocations,
+   so these tests are disabled for those targets. */
+#ifndef IOTC_PLATFORM_IS_FREERTOS
```

Comment:
This should work, doesn't it?

Reply:
No, unfortunately it doesn't though I did try it!  

We compile our FreeRTOS tests on linux and so our make system defines `IOTC_PLATFORM_BASE_POSIX`. The string `IOTC_PLATFORM_IS_FREERTOS` seems to be a makefile flag only -- it is not added to our clfags.

Then I didn't want to reuse the Makefile flag `IOTC_PLATFORM_IS_FREERTOS` in our clfags because that would add `-DIOTC_PLATFORM_IS_FREERTOS`  alongside `-DIOTC_PLATFORM_BASE_POSIX`.  Both would appear on the command line and I thought that nomenclature would be confusing, so I named the flag `IOTC_TARGET_FREERTOS`.